### PR TITLE
Added few more sample code for Search API.

### PR DESCRIPTION
**Pull request by tmatsuo**
* Moved some checkstyle modules to java-repo-tools
* Moved some lines in .gitignore to the top level

### CODE REVIEW

**Comment by tmatsuo** [comment]
@tswast @lesv I forgot to add some more code for the Search API. Also moved common configurations to the top level. Please take a look.

**Comment by tmatsuo** [comment]
Also added code on https://cloud.google.com/appengine/docs/java/search/options

File: appengine/search/src/main/java/com/example/appengine/search/DeleteServlet.java [hunk 1]
```diff
@@ -23,13 +23,13 @@
 import com.google.appengine.api.search.GetResponse;
 // [END delete_import]
 
-//CHECKSTYLE:OFF
+// CHECKSTYLE:OFF
```
Comment (lesv) [inline appengine/search/src/main/java/com/example/appengine/search/DeleteServlet.java]: Why did you remove checkstyle here?

Comment (lesv) [inline appengine/search/src/main/java/com/example/appengine/search/DeleteServlet.java]: Ah - I guess it's so that it stays in the right place w/ your comments.

**Comment by lesv** [comment]
LGTM
